### PR TITLE
Update information about saving models in the MXNet README

### PR DESCRIPTION
**Pull request by laurenyu**
...and some other tweaks. The weirdness around how I formatted the ToC title is because GH doesn't seem to render the ToC directive in a nice way (i.e. the title is just normal paragraph text).

Update: things got a little messy as I was trying to merge in master. So the Sphinx file should be basically the same as the README. The reason it looks like the entire file was changed is because the Sphinx file had Windows-style line endings (CRLF).

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have updated the [changelog](https://github.com/aws/sagemaker-python-sdk/blob/master/CHANGELOG.rst) with a description of my changes (if appropriate)
- [x] I have updated any necessary [documentation](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: doc/using_mxnet.rst [hunk 1]
```diff
@@ -1,822 +1,784 @@
...
+``mxnet-train.py``. You can run this script in SageMaker as follows:
+
+.. code:: python
+
+    from sagemaker.mxnet import MXNet
+    mxnet_estimator = MXNet('mxnet-train.py',
+                            role='SageMakerRole',
+                            train_instance_type='ml.p3.2xlarge',
+                            train_instance_count=1,
+                            framework_version='1.2.1')
```
Comment (icywang86rui) [inline doc/using_mxnet.rst]: May as well use '1.3.0'

File: doc/using_mxnet.rst [hunk 2]
```diff
@@ -1,822 +1,784 @@
...
+| For information on how to upgrade an old script to the new format, see `"Updating your MXNet training script" <#updating-your-mxnet-training-script>`__. |
++----------------------------------------------------------------------------------------------------------------------------------------------------------+
+
+For versions 1.3 and higher
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Your MXNet training script must be a Python 2.7 or 3.5 compatible source file.
+
+The training script is very similar to a training script you might run outside of SageMaker, but you can access useful properties about the training environment through various environment variables, including the following:
+
+* ``SM_MODEL_DIR``: A string that represents the path where the training job should write the model artifacts to.
```
Comment (icywang86rui) [inline doc/using_mxnet.rst]: May be we should explain this is just '/opt/ml/model' and point to the BYO document?

File: doc/using_mxnet.rst [hunk 3]
```diff
@@ -1,822 +1,784 @@
...
+  After training, artifacts in this directory are uploaded to S3 for model hosting.
+* ``SM_NUM_GPUS``: An integer representing the number of GPUs available to the host.
+* ``SM_OUTPUT_DATA_DIR``: A string that represents the path to the directory to write output artifacts to.
```
Comment (icywang86rui) [inline doc/using_mxnet.rst]: I think this is not true. I just tested this actually. Things stored in '/opt/ml/output' doesn't get uploaded. Only reference I can find about this directory is if you put anything in '/opt/ml/output/failure' will show up in the DescribeTrainingJob call response.

File: doc/using_mxnet.rst [hunk 4]
```diff
@@ -1,822 +1,784 @@
...
+
+.. code:: python
+
+    import argparse
+    import os
+
+    if __name__ == '__main__':
+        parser = argparse.ArgumentParser()
+
+        # hyperparameters sent by the client are passed as command-line argume
```
Comment (icywang86rui) [inline doc/using_mxnet.rst]: Does the requirement.txt thing still work?

Comment (icywang86rui) [inline doc/using_mxnet.rst]: I would add the hyperparamters in the example training script argparse part here.

Comment (icywang86rui) [inline doc/using_mxnet.rst]: Add the options are 'py2' and 'py3'?

Comment (icywang86rui) [inline doc/using_mxnet.rst]: This is such a bad name for what it is -_-

Comment (icywang86rui) [inline doc/using_mxnet.rst]: We can have a local file system location for this in local mode. Right?

File: doc/using_mxnet.rst (hunk 1 above)
Comment (laurenyu) [reply doc/using_mxnet.rst]: done

File: doc/using_mxnet.rst (hunk 2 above)
Comment (laurenyu) [reply doc/using_mxnet.rst]: I'm going to punt on this and let Eric decide what should point where with these kinds of things

File: doc/using_mxnet.rst (hunk 3 above)
Comment (laurenyu) [reply doc/using_mxnet.rst]: but our integ tests checking for a success file in the output dir do pass?

* https://github.com/aws/sagemaker-containers/blob/8b02b03a4edd6bfc838d955993c74a3100b121ab/src/sagemaker_containers/_files.py#L28-L34
* https://github.com/aws/sagemaker-mxnet-container/blob/master/test/integration/__init__.py#L19-L22

File: doc/using_mxnet.rst (hunk 4 above)
Comment (laurenyu) [reply doc/using_mxnet.rst]: yes. it's disabled only for TF script mode.

Comment (laurenyu) [reply doc/using_mxnet.rst]: done

Comment (laurenyu) [reply doc/using_mxnet.rst]: yeah, added.

Comment (laurenyu) [reply doc/using_mxnet.rst]: I think so. this is inaccurate across all of our docs, then. tweaked the wording here

File: src/sagemaker/mxnet/README.rst [hunk 5]
```diff
@@ -47,12 +56,9 @@ Your MXNet training script must be a Python 2.7 or 3.5 compatible source file.
 
 The training script is very similar to a training script you might run outside of SageMaker, but you can access useful properties about the training environment through various environment variables, including the following:
 
-* ``SM_MODEL_DIR``: A string that represents the path where the training job writes the model artifacts to.
+* ``SM_MODEL_DIR``: A string that represents the path where the training job should write the model artifacts to.
```
Comment (ChoiByungWook) [inline src/sagemaker/mxnet/README.rst]: should write!?!?!

Comment (laurenyu) [reply src/sagemaker/mxnet/README.rst]: lol fixed

**Pull request merged**